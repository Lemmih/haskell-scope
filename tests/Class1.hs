module Class1 where

class MyClass a where
  myClassMethod :: a -> a

fn = myClassMethod

