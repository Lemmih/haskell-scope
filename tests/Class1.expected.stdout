Scope errors:

Definitions:
  Definition 1:
    class [4mMyClass[0m a where
  Definition 2:
    class MyClass [4ma[0m where
  Definition 3:
      [4mmyClassMethod[0m :: a -> a
  Definition 4:
    [4mfn[0m = myClassMethod

Use sites:
  Definition used: 2
      myClassMethod :: [4ma[0m -> a
  Definition used: 2
      myClassMethod :: a -> [4ma[0m
  Definition used: 3
    fn = [4mmyClassMethod[0m
